tools/wptrunner/wptrunner/executors/executorwindowsaccessibility.py

```diff
@@ -0,0 +1,130 @@
+from ..executors.ia2.constants import *
+
+import json
+import sys
+import time
+
+import ctypes
+from ctypes import POINTER, byref
+from ctypes.wintypes import BOOL, HWND, LPARAM, POINT
+
+import comtypes.client
+from comtypes import COMError, IServiceProvider
+
+CHILDID_SELF = 0
+OBJID_CLIENT = -4
+
+user32 = ctypes.windll.user32
+oleacc = ctypes.oledll.oleacc
+oleaccMod = comtypes.client.GetModule("oleacc.dll")
+IAccessible = oleaccMod.IAccessible
+del oleaccMod
+
+## CoCreateInstance of UIA also initialized IA2
+uiaMod = comtypes.client.GetModule("UIAutomationCore.dll")
+uiaClient = comtypes.CoCreateInstance(
+    uiaMod.CUIAutomation._reg_clsid_,
+    interface=uiaMod.IUIAutomation,
+    clsctx=comtypes.CLSCTX_INPROC_SERVER,
+)
+
+def accessible_object_from_window(hwnd, objectID=OBJID_CLIENT):
+    p = POINTER(IAccessible)()
+    oleacc.AccessibleObjectFromWindow(
+        hwnd, objectID, byref(IAccessible._iid_), byref(p)
+    )
+    return p
+
+def name_from_hwnd(hwnd):
+    MAX_CHARS = 257
+    buffer = ctypes.create_unicode_buffer(MAX_CHARS)
+    user32.GetWindowTextW(hwnd, buffer, MAX_CHARS)
+    return buffer.value
+
+def get_browser_hwnd(product_name):
+    found = []
+
+    @ctypes.WINFUNCTYPE(BOOL, HWND, LPARAM)
+    def callback(hwnd, lParam):
+        name = name_from_hwnd(hwnd)
+        if product_name not in name.lower():
+            return True
+        found.append(hwnd)
+        return False
+
+    user32.EnumWindows(callback, LPARAM(0))
+    if not found:
+        raise LookupError(f"Couldn't find {product_name} HWND")
+    return found[0]
+
+def to_ia2(node):
+    serv = node.QueryInterface(IServiceProvider)
+    return serv.QueryService(IAccessible2._iid_, IAccessible2)
+
+def find_browser(product_name):
+    hwnd = get_browser_hwnd(product_name)
+    root = accessible_object_from_window(hwnd)
+    return to_ia2(root)
+
+def poll_for_active_tab(title, root):
+    active_tab = find_active_tab(title, root)
+    iterations = 0
+    while not active_tab:
+        time.sleep(0.01)
+        active_tab = find_active_tab(title, root)
+        iterations += 1
+
+    print(f"found active tab in {iterations} iterations", file=sys.stderr)
+    return active_tab
+
+def find_active_tab(title, root):

```

Comment:
```suggestion
def find_tab(title, root):

```
If we're searching for a tab with a specific title (or URL), then we're not really necessarily looking for the active tab.